### PR TITLE
[CLOUDTRUST-4388] Implement temporary basic identity validation

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

File: cmd/keycloakb/keycloak_bridge.go [hunk 1]
```diff
@@ -1019,6 +1019,8 @@ func main() {
 			ValidateUser:                    prepareEndpoint(kyc.MakeValidateUserEndpoint(kycComponent), "validate_user", influxMetrics, kycLogger, tracer, rateLimitKyc),
 			SendSMSConsentCode:              prepareEndpoint(kyc.MakeSendSmsConsentCodeEndpoint(kycComponent), "send_sms_consent_code", influxMetrics, kycLogger, tracer, rateLimitKyc),
 			SendSMSCode:                     prepareEndpoint(kyc.MakeSendSmsCodeEndpoint(kycComponent), "send_sms_code", influxMetrics, kycLogger, tracer, rateLimitKyc),
+
+			ValidateUserBasic: prepareEndpoint(kyc.MakeValidateUserBasicIDEndpoint(kycComponent), "basic_validate_user", influxMetrics, kycLogger, tracer, rateLimitKyc), /***TO BE REMOVED WHEN MUlTI-ACCREDITATION WILL BE IMPLEMENTED***/
```
Comment (fperot74) [inline cmd/keycloakb/keycloak_bridge.go:1023]: MUlTI / MULTI (multiple times) :-P

File: api/kyc/swagger-api_kyc.yaml [hunk 2]
```diff
@@ -439,10 +472,10 @@ components:
                 type: string
                 description: expiry date. format is DD.MM.YYYY
               expired:
-                type: bool
+                type: boolean
```
Comment (fperot74) [inline api/kyc/swagger-api_kyc.yaml:475]: Good point!

File: api/kyc/api_test.go [hunk 3]
```diff
@@ -206,13 +206,18 @@ func TestValidateUserRepresentation(t *testing.T) {
...
 		var user = createValidUser()
 		user.Attachments = nil
-		assert.Nil(t, user.Validate(), "User is expected to be valid")
+		assert.Nil(t, user.Validate(false), "User is expected to be valid")
 	})
+	t.Run("Valid users everything optional", func(t *testing.T) {
+		var user UserRepresentation
+		assert.Nil(t, user.Validate(true))
+	})
+
```
Comment (fperot74) [inline api/kyc/api_test.go:220]: We should have a test with a different result according to everythingOptional

File: pkg/kyc/component.go [hunk 4]
```diff
@@ -304,6 +307,163 @@ func (c *component) ValidateUserInSocialRealm(ctx context.Context, userID string
...
+		return err
+	}
+
+	// Store check in database
+	var validation = dto.DBCheck{
+		Operator:  &operatorName,
+		DateTime:  &now,
+		Status:    ptr("VERIFIED"),
+		Type:      ptr("IDENTITY_CHECK"),
+		Nature:    ptr("BASIC_CHECK"),
```
Comment (fperot74) [inline pkg/kyc/component.go:396]: BASIC_CHECK / basic-check ?

**Review by fperot74 — APPROVED** [review]

File: pkg/kyc/component.go (hunk 4 above)
Comment (basbeu) [reply pkg/kyc/component.go:396]: Currrently for the physical check, we set in the "Nature" field : "PHYSICAL_CHECK", even if the accreditation type is "physical-check". So I did exactly the same. Maybe @harture has an opinion about this naming.

File: pkg/kyc/authorization.go [hunk 5]
```diff
@@ -175,6 +177,23 @@ func (c *authorizationComponentMW) ValidateUser(ctx context.Context, realmName s
...
 
+/********************* (BEGIN) Temporary basic identity (TO BE REMOVED WHEN MULTI-ACCREDITATION WILL BE IMPLEMENTED) *********************/
+func (c *authorizationComponentMW) ValidateUserBasicID(ctx context.Context, userID string, user apikyc.UserRepresentation) error {
+	var action = kycValidateUserBasicID.String()
+
+	// For this method, there is no target realm provided
+	// as parameter, so we pick the current realm of the user.
+	var targetRealm = ctx.Value(cs.CtContextRealm).(string)
+
+	if err := c.authManager.CheckAuthorizationOnTargetRealm(ctx, action, targetRealm); err != nil {
```
Comment (harture) [inline pkg/kyc/authorization.go:188]: I think the authorization should be OnTargetUser

**Review by harture — APPROVED** [review]

File: pkg/kyc/authorization.go (hunk 5 above)
Comment (basbeu) [reply pkg/kyc/authorization.go:188]: The authorization is from scope realm (and this is expected). So it should be fine with CheckAuthorizationOnTargetRealm

**Pull request merged**